### ISSUE TITLE
[Bug]:Mechanoids can't parry or block (edit: only humans can parry)

### ISSUE THREAD

**Issue by skycrossercat** [CombatExtended-Continued/CombatExtended#3360]
Rimworld version: 1.5.4104
Combat Extended version: 15.6.0.0
Combat Extended source (Steam, GitHub, etc.): Dev Snapshot through discord pinned link (fresh download)
Your operating system: Windows11
Your mod list:
Harmony
Core
Royalty
Ideology
Biotech
Vanilla Expanded Framework
Combat Extended
Dead Man Switch (local test version)

**Description**
Mechs can't parry any kinds of melee attack, they either take them or dodge them, no matter how high the parry chance is.
**update: with additonal reports, insects and fleshbeasts can't parry either, it's possible that only human can parry attacks**

**Expected behavior**
They should be able to parry or block melee attacks occasionally like normal human pawns.

**To reproduce**
Spawn a scyther (you can even bump up its parry chance to extreme value through editing xml), let fully armored pawns, even with bare hands to fight it, it can't parry a single hit.
Try it again against human pawns with similar parry chance, they would possiblely parry one or two hits.

**Screenshots & log dumps**
This would be better to present with videos, but the file limit is too small.

**Complete the following checklist**
I hereby verify that I have done the following:
- [x] Confirmed that my game version and load order are correct.
- [x] Confirmed that I am running the appropriate and most updated version of Combat Extended and required compatibility patches.
- [x] Confirmed I am not running any mods with known incompatibilities with Combat Extended.
- [ ] Disabled Combat Extended and attempted to reproduce the behavior without success.  **not possible due to parry is a CE feature**

**Comment by skycrossercat** [comment]
found this block in the source code:
`if (pawn == null
                    || pawn.Dead
                    || !pawn.RaceProps.Humanlike
                    || pawn.WorkTagIsDisabled(WorkTags.Violent)
                    || !pawn.health.capacities.CapableOf(PawnCapacityDefOf.Manipulation)
                    || IsTargetImmobile(pawn)
                    || pawn.MentalStateDef == MentalStateDefOf.SocialFighting)
            {
                return false`
 and the second line states `!pawn.RaceProps.Humanlike` in this block, does that mean if pawns are not human-like, the parry code would reture to false?

**Comment by ViralReaction** [comment]
Non-Humanoid parrying has been added in. Will be added on a case by case basis [#3741 ](https://github.com/CombatExtended-Continued/CombatExtended/pull/3741)
